### PR TITLE
Dev/rm/book 16 add endpoint get weather

### PR DESCRIPTION
**Pull request by zRalphy**

### CODE REVIEW

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+            return WeatherDto.builder()
+                    .temperature(temperature.toString())
+                    .humidity(humidity.toString())
+                    .pressure(pressure.toString())
+                    .windSpeed(windSpeed.toString()).build();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        throw new IllegalStateException("Can not load weather from server.");
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:63]: this code is unreachable

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            var pressure = mainMap.get("pressure");
+            var windSpeed = windMap.get("speed");
+
+            return WeatherDto.builder()
+                    .temperature(temperature.toString())
+                    .humidity(humidity.toString())
+                    .pressure(pressure.toString())
+                    .windSpeed(windSpeed.toString()).build();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:61]: Don't hide exceptions, log it with a logger

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+public class WeatherService {
+
+    private final String hostName = "x-rapidapi-host";
+    private final String hostValue = "community-open-weather-map.p.rapidapi.com";
+    private final String keyName = "x-rapidapi-key";
+    private final String keyValue = "c72dd541f1msh23890d6e014e604p10d3c9jsn3a67da9a0d84";
+
+    public WeatherDto getWeatherByCity(String city) {
+        OkHttpClient client = new OkHttpClient();
+        String urlWeather = "https://community-open-weather-map.p.rapidapi.com/weather?q=" + city + "&units=imperial&mode=JSON";
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:32]: FYI you have used rapidapi not directly 
api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key}

but ok

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+@RequiredArgsConstructor
+public class WeatherService {
+
+    private final String hostName = "x-rapidapi-host";
+    private final String hostValue = "community-open-weather-map.p.rapidapi.com";
+    private final String keyName = "x-rapidapi-key";
+    private final String keyValue = "c72dd541f1msh23890d6e014e604p10d3c9jsn3a67da9a0d84";
+
+    public WeatherDto getWeatherByCity(String city) {
+        OkHttpClient client = new OkHttpClient();
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:31]: create client on class level (only one instance for all requests)

File: src/main/java/com/pgs/booking/service/WeatherService.java (hunk 2 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java:61]: Ok, i will use here logger.

File: src/main/java/com/pgs/booking/service/WeatherService.java (hunk 1 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java]: Yes, i will change it

File: src/main/java/com/pgs/booking/service/WeatherService.java (hunk 3 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java]: I have corrected and use it directly
api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key}

File: src/main/java/com/pgs/booking/service/WeatherService.java (hunk 4 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java]: I changed OkHttpClient to RestTemplate - outdated

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+@Getter
+@Setter
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class WeatherService {
+
+    private final String keyValue = "ee6bbf09dcc0ab10e314eda8dce3dc74";
+    private final String mode = "json";
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:24]: final static

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    private final String units = "imperial";
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:25]: final static

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    private final ObjectMapper objectMapper;
+
+    public WeatherDto getWeatherByCity(String city) {
+        RestTemplate restTemplate = new RestTemplate();
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:29]: create restTeamplate in a new Config class (see annotation @Configuration and @Bean)
and finally inject it like ObjectMapper

File: src/main/java/com/pgs/booking/service/WeatherService.java (hunk 5 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java:24]: right, I should use static

File: src/main/java/com/pgs/booking/service/WeatherService.java (hunk 6 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java:25]: right, I should use static

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.web.client.RestTemplate;
+
+@Getter
+@Setter
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class WeatherService {
+
+    private final String keyValue = "ee6bbf09dcc0ab10e314eda8dce3dc74";
```
[the remark this answers is not in the document]
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java:23]: ok, I will change it

[the remark this answers is not in the document]
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java:23]: I'll try to do it this way

File: src/main/java/com/pgs/booking/service/WeatherService.java (hunk 7 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java]: ok, i will change it

File: src/main/java/com/pgs/booking/service/WeatherService.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class WeatherService {
+    @Value("${openwaether.api-key}")
+    private String keyValue;
+    private static final String mode = "json";
+    private static final String units = "imperial";
+    private final ObjectMapper objectMapper;
+    @Autowired
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/service/WeatherService.java:29]: do you need that at is autowired via constructor

Comment (zRalphy) [reply src/main/java/com/pgs/booking/service/WeatherService.java]: I have the final field with one constructor, so I do not need it

**Review by gstudzinski — APPROVED** [review]
approved

**Pull request merged**